### PR TITLE
Fix: QuantitySelector validation when min/max quantity 

### PR DESCRIPTION
**Pull request by rayssamfreitas**
## What's the purpose of this pull request?

Fix an issue with the quantity selector where, for example, when the minimum quantity was 10, it was not possible to delete this value to enter another.

## How it works?

Previously, validation occurred immediately after the input value was changed. Now, instead, validation occurs on focus out.

## How to test it?

Adding a cart with minimum quantity and trying to replace the value

### CODE REVIEW

**Review by eduardoformiga — COMMENTED** [review]
Hey, @rayssamfreitas. Thank you again for the collaboration ⭐ 

The code looks good to me! I'll ask the team to review as well :) 

I validated the behavior internally with the FastStore team, and they suggested a `Toast` message when the inputted value is out of bounds (greater than the max or less than the min). 

So, I took the liberty of making some changes here in your PR directly, hahaha

Now we have the Toast.
Screenshot 2024-07-03 at 19 41 07

What do you think?

File: packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx [hunk 1]
```diff
@@ -70,47 +77,45 @@ const QuantitySelector = ({
...
+      return roundedQuantity / unitMultiplier
+    })
   }
 
-  
   useEffect(() => {
     initial && setQuantity(initial)
   }, [initial])
 
+  const changeValueInput = (e: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (hellofanny) [inline packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx:115]:
```suggestion
  const changeInputValue = (e: React.ChangeEvent<HTMLInputElement>) => {
```

File: packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx [hunk 2]
```diff
@@ -132,7 +137,7 @@ const QuantitySelector = ({
         id="quantity-selector-input"
         aria-label="Quantity"
         value={useUnitMultiplier ? multipliedUnit : quantity}
-        onChange={validateInput}
+        onChange={changeValueInput}
```
Comment (eduardoformiga) [inline packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx:140]:
```suggestion
        onChange={changeInputValue}
```

**Comment by rayssamfreitas** [comment]
Thanks @eduardoformiga ! I think the toast idea was nice!

**Review by eduardoformiga — APPROVED** [review]

**Review by hellofanny — APPROVED** [review]

**Pull request merged**